core/error.rs

```diff
@@ -773,6 +923,410 @@ pub fn throw_type_error(scope: &mut v8::HandleScope, message: impl AsRef<str>) {
+            .into(),
+            v8::PropertyAttribute::DONT_DELETE
+            | v8::PropertyAttribute::DONT_ENUM
+            | v8::PropertyAttribute::READ_ONLY,
+          );
+        }
+      )+
+    };
+  }
+
+  // delegate to the original callsite object
+  make_delegate!(
+    scope,
+    template,
+    [
+      // excludes getFileName, which we'll override below
+      GET_THIS,
+      GET_TYPE_NAME,
+      GET_FUNCTION,
+      GET_FUNCTION_NAME,
+      GET_METHOD_NAME,
+      GET_LINE_NUMBER,
+      GET_COLUMN_NUMBER,
+      GET_EVAL_ORIGIN,
+      IS_TOPLEVEL,
+      IS_EVAL,
+      IS_NATIVE,
+      IS_CONSTRUCTOR,
+      IS_ASYNC,
+      IS_PROMISE_ALL,
+      GET_PROMISE_INDEX,
+    ]
+  );
+
+  let get_file_name_key = GET_FILE_NAME.v8_string(scope).into();
+  template.set_with_attr(
+    get_file_name_key,
+    v8::FunctionBuilder::<v8::FunctionTemplate>::new(
+      |scope: &mut v8::HandleScope<'_>,
+       args: v8::FunctionCallbackArguments<'_>,
+       mut rv: v8::ReturnValue<'_>| {
+        // lookup the original call site object
+        let orig_key = original_call_site_key(scope);
+        let orig = args.this().get_private(scope, orig_key).unwrap().cast();
+        // call getFileName
+        let orig_ret =
+          call_method::<v8::String>(scope, orig, GET_FILE_NAME, &[]);
+        if let Some(ret_val) = orig_ret {
+          // strip off `file://`
+          let string = ret_val.to_rust_string_lossy(scope);
+          let file_name = if string.starts_with("file://") {
+            Url::parse(&string)
+              .unwrap()
+              .to_file_path()
+              .unwrap()
+              .to_string_lossy()
+              .into_owned()
+          } else {
+            string
+          };
+          let v8_str =
+            crate::FastString::from(file_name).v8_string(scope).into();
+          rv.set(v8_str);
+        }
+      },
+    )
+    .build(scope)
+    .into(),
+    v8::PropertyAttribute::DONT_DELETE
+      | v8::PropertyAttribute::DONT_ENUM
+      | v8::PropertyAttribute::READ_ONLY,
+  );
+
+  template
+}
+
+// called by V8 whenever a stack trace is created.
+// note that this is called _instead_ of `Error.prepareStackTrace`
+// so we have to explicitly call it to keep user code wrking
+pub fn prepare_stack_trace_callback<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  error: v8::Local<'s, v8::Value>,
+  callsites: v8::Local<'s, v8::Array>,
+) -> v8::Local<'s, v8::Value> {
+  // stash the callsites on the error object. this is used by `JsError::from_exception_inner`
+  // to get more details on an error, because the v8 StackFrame API is missing some info.
+  if let Ok(obj) = error.try_cast::<v8::Object>() {
+    let key = call_site_evals_key(scope);
+    obj.set_private(scope, key, callsites.into());
+  }
+
+  // `globalThis.Error.prepareStackTrace`
+  let global = scope.get_current_context().global(scope);
+  let global_error = get_property(scope, global, ERROR).unwrap().cast();
```

Comment:
I wonder if we could get it only once on startup and store it. Maybe not a big deal